diff -up gnome-software-47.0/meson.build.without-help gnome-software-47.0/meson.build
--- gnome-software-47.0/meson.build.without-help	2024-09-16 12:08:30.314978903 +0200
+++ gnome-software-47.0/meson.build	2024-09-16 12:08:44.681170496 +0200
@@ -314,7 +314,6 @@ if get_option('external_appstream')
 endif
 subdir('po')
 subdir('doc')
-subdir('help')
 
 # FIXME: remove when https://github.com/mesonbuild/meson/issues/837 fixed
 meson.add_install_script('meson_post_install.sh')
diff -up gnome-software-47.0/plugins/epiphany/gs-plugin-epiphany.c.without-help gnome-software-47.0/plugins/epiphany/gs-plugin-epiphany.c
--- gnome-software-47.0/plugins/epiphany/gs-plugin-epiphany.c.without-help	2024-09-16 12:11:54.253698750 +0200
+++ gnome-software-47.0/plugins/epiphany/gs-plugin-epiphany.c	2024-09-16 12:12:11.827933131 +0200
@@ -413,8 +413,6 @@ gs_epiphany_refine_app_state (GsPlugin *
 			gs_app_set_state (app, GS_APP_STATE_AVAILABLE);
 		else {
 			gs_app_set_state (app, GS_APP_STATE_UNAVAILABLE);
-			gs_app_set_url_missing (app,
-						"help:gnome-software/how-to-reinstall-a-web-app");
 		}
 	}
 }
diff -up gnome-software-47.0/src/gs-age-rating-context-dialog.ui.without-help gnome-software-47.0/src/gs-age-rating-context-dialog.ui
--- gnome-software-47.0/src/gs-age-rating-context-dialog.ui.without-help	2024-09-16 12:13:52.655277823 +0200
+++ gnome-software-47.0/src/gs-age-rating-context-dialog.ui	2024-09-16 12:13:56.668331343 +0200
@@ -56,6 +56,8 @@
         </child>
         <child>
           <object class="AdwPreferencesGroup">
+            <property name="visible">False</property>
+            <property name="sensitive">False</property>
             <child>
               <object class="AdwButtonRow">
                 <property name="title" translatable="yes">_How to contribute missing information</property>
diff -up gnome-software-47.0/src/gs-details-page.c.without-help gnome-software-47.0/src/gs-details-page.c
--- gnome-software-47.0/src/gs-details-page.c.without-help	2024-09-16 12:22:07.020871002 +0200
+++ gnome-software-47.0/src/gs-details-page.c	2024-09-16 12:22:56.795534829 +0200
@@ -626,27 +626,18 @@ gs_details_page_license_tile_get_involve
 			uri = gs_app_get_url (self->app, AS_URL_KIND_HOMEPAGE);
 	} else {
 		if (gs_app_get_license (self->app) == NULL) {
-			uri = "help:gnome-software/software-metadata#license";
+			/* would show help URI */
 		} else {
 			license_url = as_get_license_url (gs_app_get_license (self->app));
 
 			if (license_url != NULL && *license_url != '\0') {
 				uri = license_url;
-			} else {
-				/* Page to explain the differences between FOSS and proprietary
-				 * software. This is a page on the gnome-software wiki for now,
-				 * so that we can update the content independently of the release
-				 * cycle. Likely, we will link to a more authoritative source
-				 * to explain the differences.
-				 * Ultimately, we could ship a user manual page to explain the
-				 * differences (so that it’s available offline), but that’s too
-				 * much work for right now. */
-				uri = "help:gnome-software/software-licensing";
 			}
 		}
 	}
 
-	gs_shell_show_uri (self->shell, uri);
+	if (uri != NULL)
+		gs_shell_show_uri (self->shell, uri);
 }
 
 static void
diff -up gnome-software-47.0/src/gs-hardware-support-context-dialog.ui.without-help gnome-software-47.0/src/gs-hardware-support-context-dialog.ui
--- gnome-software-47.0/src/gs-hardware-support-context-dialog.ui.without-help	2024-09-16 12:23:52.542278301 +0200
+++ gnome-software-47.0/src/gs-hardware-support-context-dialog.ui	2024-09-16 12:24:00.332382197 +0200
@@ -58,6 +58,8 @@
         </child>
         <child>
           <object class="AdwPreferencesGroup">
+            <property name="visible">False</property>
+            <property name="sensitive">False</property>
             <child>
               <object class="AdwButtonRow">
                 <property name="title" translatable="yes">_How to contribute missing information</property>
diff -up gnome-software-47.0/src/gs-license-tile.c.without-help gnome-software-47.0/src/gs-license-tile.c
--- gnome-software-47.0/src/gs-license-tile.c.without-help	2024-09-16 12:19:52.028070653 +0200
+++ gnome-software-47.0/src/gs-license-tile.c	2024-09-16 12:20:58.089951698 +0200
@@ -150,7 +150,7 @@ gs_license_tile_refresh (GsLicenseTile *
 		lozenge_icon_names[0] = "community-blocked-symbolic";
 		lozenge_icon_names[1] = "dialog-warning-symbolic";
 		lozenge_icon_names[2] = "software-license-symbolic";
-		get_involved_visible = TRUE;
+		get_involved_visible = FALSE;
 		get_involved_label = _("_Learn More");
 
 		if (is_application) {
@@ -168,7 +168,7 @@ gs_license_tile_refresh (GsLicenseTile *
 		lozenge_icon_names[0] = "community-blocked-symbolic";
 		lozenge_icon_names[1] = "proprietary-code-symbolic";
 		lozenge_icon_names[2] = "software-license-symbolic";
-		get_involved_visible = TRUE;
+		get_involved_visible = FALSE;
 		get_involved_label = _("_Learn More");
 
 		if (is_application) {
@@ -218,7 +218,7 @@ gs_license_tile_refresh (GsLicenseTile *
 							 license_url,
 							 license_spdx);
 		} else {
-			get_involved_visible = TRUE;
+			get_involved_visible = FALSE;
 			/* Translators: The placeholder here is the name of a software license. */
 			description = g_strdup_printf (_("This software is developed under the special license “%s”."
 							  "\n\n"
diff -up gnome-software-47.0/src/gs-safety-context-dialog.ui.without-help gnome-software-47.0/src/gs-safety-context-dialog.ui
--- gnome-software-47.0/src/gs-safety-context-dialog.ui.without-help	2024-09-16 12:24:27.198740507 +0200
+++ gnome-software-47.0/src/gs-safety-context-dialog.ui	2024-09-16 12:25:48.974831126 +0200
@@ -121,6 +121,8 @@
 
         <child>
           <object class="AdwPreferencesGroup">
+            <property name="visible">False</property>
+            <property name="sensitive">False</property>
             <child>
               <object class="AdwButtonRow">
                 <property name="title" translatable="yes">_How to contribute missing information</property>
